### PR TITLE
asserts: refine Backstore interface passing primaryKey header names in

### PR DESCRIPTION
**Pull request by pedronis**
based on #302

### CODE REVIEW

File: asserts/database.go [hunk 1]
```diff
@@ -29,29 +29,21 @@ import (
...
+	// Put stores an assertion for the given primaryKey field names forming a unique key.
 	// It is responsible for checking that assert is newer than a
 	// previously stored revision.
-	Put(assertType AssertionType, primaryPath []string, assert Assertion) error
-	// Get loads an assertion with the given unique primaryPath.
+	Put(assertType AssertionType, primaryKey []string, assert Assertion) error
+	// Get loads the assertion with the given unique keyValues for the primaryKey fields.
 	// If none is present it returns ErrNotFound.
-	Get(assertType AssertionType, primaryPath []string) (Assertion, error)
+	Get(assertType AssertionType, primaryKey, keyValues []string) (Assertion, error)
```
Comment (emgee) [inline asserts/database.go]: I don't find `primaryKey` totally obvious. Maybe s/primaryKey/keyHeaders and s/keyValues/key ?

**Comment by emgee** [comment]
+1

File: asserts/database.go [hunk 2]
```diff
@@ -30,22 +30,19 @@ import (
 )
 
 // Backstore is a backstore for assertions. It can store and retrieve
-// assertions by type under primary paths (tuples of strings). Plus it
-// supports more general searches.
+// assertions by type under unique primary key headers. Plus it supports searching by headers.
 type Backstore interface {
-	// Put stores an assertion under the given unique primaryPath.
+	// Put stores an assertion for the given primaryKeyHeaders forming a unique key.
```
Comment (chipaca) [inline asserts/database.go]: can't we just use `Put` as the verb in that sentence? `Put stores ...` reads wrong.

File: asserts/database.go [hunk 3]
```diff
@@ -30,22 +30,19 @@ import (
 )
 
 // Backstore is a backstore for assertions. It can store and retrieve
```
Comment (chipaca) [inline asserts/database.go]: Can this be `Backstore for assertions`? or is there a convention about having a verb there that we must follow?

File: asserts/database.go [hunk 4]
```diff
@@ -30,22 +30,19 @@ import (
...
+// assertions by type under unique primary key headers. Plus it supports searching by headers.
 type Backstore interface {
-	// Put stores an assertion under the given unique primaryPath.
+	// Put stores an assertion for the given primaryKeyHeaders forming a unique key.
 	// It is responsible for checking that assert is newer than a
 	// previously stored revision.
-	Put(assertType AssertionType, primaryPath []string, assert Assertion) error
-	// Get loads an assertion with the given unique primaryPath.
+	Put(assertType AssertionType, primaryKeyHeaders []string, assert Assertion) error
+	// Get loads the assertion with the given unique key for the primaryKeyHeaders.
```
Comment (chipaca) [inline asserts/database.go]: again I don't see the need for the extra verb; "Get the assertion" works fine (and the fact that it works fine means it's the right name for the method).

File: asserts/database.go [hunk 5]
```diff
@@ -30,22 +30,19 @@ import (
...
 	// If none is present it returns ErrNotFound.
-	Get(assertType AssertionType, primaryPath []string) (Assertion, error)
+	Get(assertType AssertionType, primaryKeyHeaders, key []string) (Assertion, error)
 	// Search searches for assertions matching the given headers.
```
Comment (chipaca) [inline asserts/database.go]: augh

**Comment by chipaca** [comment]
Some silly obsessing with comments, but otherwise :+1:

**Pull request merged**